mmeval/metrics/bleu.py

```diff
@@ -0,0 +1,188 @@
+    The closer the translation is to human translation,
+    the higher the score will be.
+
+    Args:
+        n_gram (int): The maximum number of words contained in a phrase
+            when calculating word fragments. Defaults to 4.
+        smooth(bool): Whether or not to apply smoothing. Default to False.
+        ngram_weights(Sequence[float], optional): Weights used
+            for unigrams, bigrams, etc. to calculate BLEU score.
+            If not provided, uniform weights are used. Default to None.
+        **kwargs: Keyword parameters passed to :class:`BaseMetric`.
+
+    Examples:
+
+        >>> predictions = ['the cat is on the mat', 'There is a big tree near the park here']  # noqa: E501
+        >>> references = [['a cat is on the mat'], ['A big tree is growing near the park here']]  # noqa: E501
+        >>> bleu = Bleu()
+        >>> bleu_results = bleu(predictions, references)
+        {'bleu': ...}
+
+    Calculate Bleu with smooth:
+
+        >>> predictions = ['the cat is on the mat', 'There is a big tree near the park here']  # noqa: E501
+        >>> references = [['a cat is on the mat'], ['A big tree is growing near the park here']]  # noqa: E501
+        >>> bleu = Bleu(smooth = True)
+        >>> bleu_results = bleu(predictions, references)
+        {'bleu': ...}
+    """
+
+    def __init__(self,
+                 n_gram: int = 4,
+                 smooth: bool = False,
+                 ngram_weights: Optional[Sequence[float]] = None,
+                 **kwargs) -> None:
+        super().__init__(**kwargs)
+        self.n_gram = n_gram
+        self.smooth = smooth
+        if ngram_weights is not None and len(ngram_weights) != n_gram:
+            raise ValueError(
+                f'List of weights has different weights than `n_gram`: '
+                f'{len(ngram_weights)} != {n_gram}')
+        if ngram_weights is None:
+            ngram_weights = [1.0 / n_gram] * n_gram
+        self.ngram_weights = ngram_weights
+
+    def add(self, predictions: Sequence[str], references: Sequence[Sequence[str]]) -> None:  # type: ignore # yapf: disable # noqa: E501
+        """Add the intermediate results to ``self._results``.
+
+        Args:
+             predictions (Sequence[str]): An iterable of machine
+                translated corpus.
+             references (Sequence[Sequence[str]]): An iterable of
+                iterables of reference corpus.
+        """
+
+        references_token: Sequence[Sequence[Sequence[str]]] = [[
+            tokenizer_fn(line) if line is not None else [] for line in r
+        ] for r in references]
+        predictions_token: Sequence[Sequence[str]] = [
+            tokenizer_fn(line) if line else [] for line in predictions
+        ]
+        for prediction, references in zip(predictions_token, references_token):
+            pred_len = len(prediction)
+            references_len = len(
+                min(references, key=lambda x: abs(len(x) - pred_len)))
```

Comment:
It is better to add the inline comment~
```suggestion
            pred_len = len(prediction)
            # Find the reference that is closest in length to the prediction
            references_len = len(
                min(references, key=lambda x: abs(len(x) - pred_len)))
```